### PR TITLE
Updates sprite widget to handle changes to renderbox

### PR DESCRIPTION
**Pull request by vlidholt**
Updates sprite widget to handle changes to renderbox
Adds main screen to game and graphics for game UI

### CODE REVIEW

File: sky/sdk/example/game/assets/game_ui.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
+{"frames": [
+
+{
+	"filename": "bar_shield.png",
```
Comment (eseidelGoogle) [inline sky/sdk/example/game/assets/game_ui.json]: Tabs?

File: sky/sdk/example/game/lib/sprite_box.dart [hunk 2]
```diff
@@ -28,6 +28,20 @@ class SpriteBox extends RenderBox {
   // Root node for drawing
   NodeWithSize _rootNode;
 
+  void set rootNode (NodeWithSize value) {
+    if (value == _rootNode)
+      return;
+
+    // Remove sprite box references
+    if (_rootNode != null) _removeSpriteBoxReference(_rootNode);
```
Comment (eseidelGoogle) [inline sky/sdk/example/game/lib/sprite_box.dart]: A bit odd that the one-line if above and this one use different indent styles.

File: sky/sdk/example/game/lib/sprite_box.dart [hunk 3]
```diff
@@ -76,6 +105,16 @@ class SpriteBox extends RenderBox {
     }
   }
 
+  void attach() {
+    super.attach();
+    _scheduleTick();
+  }
+
+  void detach() {
```
Comment (Hixie) [inline sky/sdk/example/game/lib/sprite_box.dart]: remove the override if you're not using it

File: sky/sdk/example/game/lib/sprite_box.dart [hunk 4]
```diff
@@ -36,6 +50,15 @@ class SpriteBox extends RenderBox {
   // Transformation mode
   SpriteBoxTransformMode _transformMode;
 
+  void set transformMode (SpriteBoxTransformMode value) {
+    if (value == _transformMode)
+      return;
+    _transformMode = value;
+
+    // Invalidate stuff
+    if (attached) performLayout();
```
Comment (eseidelGoogle) [inline sky/sdk/example/game/lib/sprite_box.dart]: Unless if (foo) return; is special, seems we should pick a one line if style and stick to it.  @Hixie likely already did.

File: sky/sdk/example/game/lib/sprite_box.dart [hunk 5]
```diff
@@ -76,6 +105,16 @@ class SpriteBox extends RenderBox {
...
   }
 
+  void attach() {
+    super.attach();
+    _scheduleTick();
+  }
+
+  void detach() {
+    super.detach();
+
```
Comment (eseidelGoogle) [inline sky/sdk/example/game/lib/sprite_box.dart]: Is this even needed?

File: sky/sdk/example/game/main.dart [hunk 6]
```diff
@@ -54,9 +78,22 @@ class GameDemoApp extends App {
...
+    return new SpriteWidget(
+      new GameDemoWorld(_app, navigator, _loader, _spriteSheet)
+    );
+  }
+
+  Widget _buildMainScene(navigator, route) {
+    return new Center(
+      child: new RaisedButton(
+        child: new Text("Play"),
+        onPressed:() => navigator.pushNamed('/game')
```
Comment (eseidelGoogle) [inline sky/sdk/example/game/main.dart]: Typically a space after :

**Comment by eseidelGoogle** [comment]
lgtm with nits fixed.

File: sky/sdk/example/game/lib/sprite_box.dart (hunk 2 above)
Comment (vlidholt) [inline sky/sdk/example/game/lib/sprite_box.dart]: Different authors :P

File: sky/sdk/example/game/lib/sprite_box.dart (hunk 5 above)
Comment (vlidholt) [inline sky/sdk/example/game/lib/sprite_box.dart]: Removing

File: sky/sdk/example/game/assets/game_ui.json (hunk 1 above)
Comment (vlidholt) [inline sky/sdk/example/game/assets/game_ui.json]: This is an auto-generated file, by a texture packer. It's not meant to be edited manually.

File: sky/sdk/example/game/lib/sprite_box.dart (hunk 3 above)
Comment (vlidholt) [inline sky/sdk/example/game/lib/sprite_box.dart]: Fixed

File: sky/sdk/example/game/main.dart (hunk 6 above)
Comment (vlidholt) [inline sky/sdk/example/game/main.dart]: Fixing

**Pull request merged**